core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +122,16 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  public static Integer existingDataFiles(Snapshot snapshot) {
```

Comment:
Can you add `Count` to the method name? I think it would be more clear what is being accumulated here.